cli/azd/pkg/infra/provisioning/terraform/terraform_provider.go

```diff
@@ -0,0 +1,559 @@
+	if _, err := os.Stat(parametersFilePath); err != nil {
+		err = t.createParametersFile()
+		if err != nil {
+			return nil, fmt.Errorf("creating parameters file: %w", err)
+		}
+	}
+
+	parametersBytes, err := os.ReadFile(parametersFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("reading parameter file template: %w", err)
+	}
+
+	if err := json.Unmarshal(parametersBytes, &parameters); err != nil {
+		return nil, fmt.Errorf("error unmarshalling template parameters: %w", err)
+	}
+
+	for key, param := range parameters {
+		templateParameters[key] = InputParameter{
+			Type:  key,
+			Value: param,
+		}
+	}
+
+	template := Deployment{
+		Parameters: templateParameters,
+	}
+
+	return &template, nil
+}
+
+// Copies the Terraform parameters file from the project template into the .azure environment folder
+func (t *TerraformProvider) createParametersFile() error {
+	// Copy the parameter template file to the environment working directory and do substitutions.
+	parametersTemplateFilePath := t.parametersTemplateFilePath()
+
+	log.Printf("Reading parameters template file from: %s", parametersTemplateFilePath)
+	parametersBytes, err := os.ReadFile(parametersTemplateFilePath)
+	if err != nil {
+		return fmt.Errorf("reading parameter file template: %w", err)
+	}
+	replaced, err := envsubst.Eval(string(parametersBytes), func(name string) string {
+		if val, has := t.env.Values[name]; has {
+			return val
+		}
+		return os.Getenv(name)
+	})
+
+	if err != nil {
+		return fmt.Errorf("substituting parameter file: %w", err)
+	}
+
+	parametersFilePath := t.parametersFilePath()
+	writeDir := filepath.Dir(parametersFilePath)
+	if err := os.MkdirAll(writeDir, 0755); err != nil {
+		return fmt.Errorf("creating directory structure: %w", err)
+	}
+
+	log.Printf("Writing parameters file to: %s", parametersFilePath)
+	err = os.WriteFile(parametersFilePath, []byte(replaced), 0644)
+	if err != nil {
+		return fmt.Errorf("writing parameter file: %w", err)
+	}
+
+	return nil
+}
+
+// Copies the Terraform backend file from the project template into the .azure environment folder
+func (t *TerraformProvider) createBackendConfigFile() error {
+	// Copy the backend config template file to the environment working directory and do substitutions.
+	backendTemplateFilePath := t.backendConfigTemplateFilePath()
+
+	log.Printf("Reading backend config template file from: %s", backendTemplateFilePath)
+	backendBytes, err := os.ReadFile(backendTemplateFilePath)
+	if err != nil {
+		return fmt.Errorf("reading backend config file template: %w", err)
+	}
+	replaced, err := envsubst.Eval(string(backendBytes), func(name string) string {
+		if val, has := t.env.Values[name]; has {
+			return val
+		}
+		return os.Getenv(name)
+	})
+
+	if err != nil {
+		return fmt.Errorf("substituting backend config file: %w", err)
+	}
+
+	backendConfigFilePath := t.backendConfigFilePath()
+	writeDir := filepath.Dir(backendConfigFilePath)
+	if err := os.MkdirAll(writeDir, 0755); err != nil {
```

Comment:
Please use our `osutil` package for permission setting.
```suggestion
	if err := os.MkdirAll(writeDir, osutil.PermissionDirectory); err != nil {
```

Reply:
fixed